src/Forms/SelectField.php

```diff
@@ -241,7 +241,7 @@ public function isSelectedValue($dataValue, $userValue)
     public function performReadonlyTransformation()
     {
         /** @var LookupField $field */
-        $field = $this->castedCopy('SilverStripe\\Forms\\LookupField');
```

Comment:
I'm going to be mean and state that you should have separate performReadonlyTransformation in both SingleSelectField and MultiSelectField (both direct subclasses of SelectField). You've switched the assumption from "all select fields use the multi select readonly" to "all select fields use the single select readonly".